pkg/cryptutil/hash.go

```diff
@@ -28,3 +29,14 @@ func HashPassword(password []byte) ([]byte, error) {
 func CheckPasswordHash(hash, password []byte) error {
 	return bcrypt.CompareHashAndPassword(hash, password)
 }
+
+// HashProto hashes a protobuf message. It sets `Deterministic` to true to ensure
+// the encoded message is always the same. (ie map order is lexographic)
+func HashProto(msg proto.Message) []byte {
```

Comment:
Could we add tests to this func?

Reply:
I added a test but it will rarely succeed in the non deterministic case too.